### PR TITLE
Cleanup usage of base::Optional and base::nullopt #16589

### PR DESCRIPTION
**Pull request by mariospr**
These types are deprecated and we should be using absl::optional and
absl::nullopt now and avoiding the old ones in newly added code, so that
we don't have to keep changing that when rebasing the cr93 branch.

Resolves https://github.com/brave/brave-browser/issues/16589

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Comment by mariospr** [comment]
@mkarolin @darkdh With cr92 merged into master, I think this is probably a good moment to merge this change, PTAL thanks!

@bbondy JFYI

File: components/brave_vpn/brave_vpn_service.cc [hunk 1]
```diff
@@ -7,6 +7,7 @@
 
 #include "base/json/json_reader.h"
 #include "base/json/json_writer.h"
+#include "third_party/abseil-cpp/absl/types/optional.h"
```
Comment (darkdh) [inline components/brave_vpn/brave_vpn_service.cc:10]: shouldn't we add dependency for `//third_party/abseil-cpp:absl` in `components/brave_vpn/BUILD.gn`?

Comment (mariospr) [reply components/brave_vpn/brave_vpn_service.cc:10]: Oops! Indeed, added now. Also, made the most of this force-push to remove this include from chromium_src overrides where the file being overriden already includes it, and remove a temporary include rule from src/DEPs that's not needed anymore.

PTAL and thanks again

**Comment by mariospr** [comment]
@darkdh @mkarolin  Pinging again on this, thanks in advance!

**Review by mkarolin — APPROVED** [review]
++

**Comment by mariospr** [comment]
@iefremov PTAL to the trivial change in the DEPS file (remove a temporary include rule that's no longer needed), thanks!

**Review by darkdh — APPROVED** [review]

**Review by iefremov — APPROVED** [review]

**Pull request merged**